python/cudf/cudf/core/multiindex.py

```diff
@@ -1015,7 +1016,11 @@ def __getitem__(self, index):
         elif isinstance(index, slice):
             start, stop, step = index.indices(len(self))
             index = column.arange(start, stop, step)
-        result = MultiIndex.from_frame(self.to_frame(index=False).take(index))
+        result = MultiIndex.from_frame(
+            self.to_frame(index=False, name=list(range(0, self.nlevels))).take(
```

Comment:
Should existing names be used instead of `range(...)`?

Reply:
Actually nope, since names are being assigned below. Since name assignment is far below I moved it near to this function call.